hypervisor/debug/shell.c

```diff
@@ -539,29 +538,30 @@ static int32_t shell_cmd_help(__unused int32_t argc, __unused char **argv)
 			}
 
 			i++;
+			if (p_cmd->cmd_param == NULL)
+				p_cmd->cmd_param = " ";
+			(void)memset(str, ' ', sizeof(str));
+			/* Output the command & parameter string */
+			snprintf(str, MAX_OUTPUT_LEN, " %-15s%-64s",
+					p_cmd->str, p_cmd->cmd_param);
+			shell_puts(str);
+			shell_puts("\r\n");
 
-			/* Output the command string */
-			shell_puts("  ");
-			shell_puts(p_cmd->str);
-
-			/* Calculate spaces needed for alignment */
-			spaces = MAX_INDENT_LEN - strnlen_s(p_cmd->str, MAX_INDENT_LEN - 1);
-
-			space_buf[spaces] = '\0';
-			shell_puts(space_buf);
-			space_buf[spaces] = ' ';
-
-			/* Display parameter info if applicable. */
-			if (p_cmd->cmd_param != NULL) {
-				shell_puts(p_cmd->cmd_param);
-			}
-
-			/* Display help text if available. */
-			if (p_cmd->help_str != NULL) {
-				shell_puts(" - ");
-				shell_puts(p_cmd->help_str);
+			help_str = p_cmd->help_str;
+			while (strnlen_s(help_str, MAX_OUTPUT_LEN > 0)) {
+				(void)memset(str, ' ', sizeof(str));
+				if (strnlen_s(help_str, MAX_OUTPUT_LEN) > 65) {
+					snprintf(str, MAX_OUTPUT_LEN, "               %-s", help_str);
+					shell_puts(str);
+					shell_puts("\r\n");
+					help_str = help_str + 65;
```

Comment:
So this is just breaking the help_str at 65-character boundaries, not at a word boundary, eh?  So it will put part of a word on one line and the rest on the next.  Can you show us what this help output looks like now?

Reply:
@dbkinder Yes, this still have the issue will break the word, the shell_puts only can output via character length.

Reply:
![image](https://user-images.githubusercontent.com/36183357/133709050-5618c148-10c3-41a4-b09b-9537eb965780.png)

Reply:
![image](https://user-images.githubusercontent.com/36183357/133709127-196051e3-ca44-4c6b-9354-92444fdd39ce.png)

Reply:
It looks like one character is missing from a word when the line breaks.

Reply:
We can merge this now and fix later.